mondrian-generator.js

```diff
@@ -48,41 +60,74 @@ function composition() {
   y.unshift(2);
   y.push(398);
 
-  var canvas = document.getElementById('compositionCanvas');
-  var context = canvas.getContext('2d');
+  return { x, y }
+}
+
+function getContext() {
+  const canvas = document.getElementById('compositionCanvas');
+  const context = canvas.getContext('2d');
+  return { context, canvas }
+}
 
-  // vertical lines
-  for (i = 0; i < x.length; i++) {
-    context.beginPath();
-    context.moveTo(x[i], 0);
-    context.lineTo(x[i], 400);
-    context.stroke();
-    context.lineWidth = 4;
-    context.strokeStyle = 'black';
-    context.stroke();
-  };
-  // horizontal lines
-  for (j = 0; j < y.length; j++) {
+
+function getLineWidth(i, linePositions) {
+  // const LINE_WIDTHS = [2, 4, 6, 8, 10]
+  const LINE_WIDTHS = [4]
+  const DEFAULT_WIDTH = 4
+  const LINE_WIDTH = i === 0 || i === linePositions.length - 1 ? DEFAULT_WIDTH : LINE_WIDTHS[getRandomInt(0, LINE_WIDTHS.length)]
+  return LINE_WIDTH
+}
+
+function addLinesToContext(context, linePositions, xOrY) {
```

Comment:
Condenses code that was previously repeated for both x and y into a single function.